File: loop/DoublingTest.dart
int count(List<int> a) {
  int N = a.length;
  int cnt = 0;
  for (int i = 0; i < N; i++) {
    for (int j = i + 1; j < N; j++) {
      for (int k = j + 1; k < N; k++) {
        if (a[i] + a[j] + a[k] == 0) {
          cnt++;
        }
      }
    }
  }
  return cnt;
}
int timeTrial(int N) {
  var a = new List<int>(N);
  for (int i = 0; i < N; i++) {
    a[i] = i + 1;
  }
  return count(a);
}

void main() {
  int res;
  res = timeTrial(250);
  res = timeTrial(500);
  res = timeTrial(1000);
  res = timeTrial(2000);
}

File: loop/doublingTest.go
package main

func Count(a []int) int {
	n := len(a)
	cnt := 0
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			for k := j + 1; k < n; k++ {
				if a[i]+a[j]+a[k] == 0 {
					cnt++
				}
			}
		}
	}
	return cnt
}

func timeTrial(n int) int {
	a := make([]int, n)
	for i := 0; i < n; i++ {
		a[i] = i + 1
	}
	return Count(a)
}
func main() {
	res := timeTrial(250)
	res = timeTrial(500)
	res = timeTrial(1000)
	res = timeTrial(2000)
	_ = res
}
